subprojects/core/src/main/java/org/gradle/api/internal/file/copy/DefaultFileCopyDetails.java

```diff
@@ -215,7 +222,15 @@ public ContentFilterable filter(Class<? extends FilterReader> filterType) {
 
     @Override
     public ContentFilterable expand(Map<String, ?> properties) {
-        filterChain.expand(properties);
+        filterChain.expand(properties, Providers.notDefined());
+        return this;
+    }
+
+    @Override
+    public ContentFilterable expand(Map<String, ?> properties, Action<? super ExpandDetails> action) {
+        ExpandDetails details = objectFactory.newInstance(ExpandDetailsImpl.class);
```

Comment:
You don't need an implementation of this class, you should be able to just do:
```suggestion
        ExpandDetails details = objectFactory.newInstance(ExpandDetails.class);
        details.getEscapeBackslash().convention(false);
```

Gradle will generate an implementation for simple things like this.

Reply:
Thanks, I wasn't aware of this.